resources/client/cl_main.ts

```diff
@@ -180,3 +180,25 @@ RegisterNuiCB<{ keepGameFocus: boolean }>(
 //   const time = getCurrentGameTime()
 //   sendMessage('PHONE', 'setTime', time)
 // }, 2000);
+
+// Set a global function to check if the player is loaded
+let playerLoaded = false;
+(global as any).playerReady = async () => {
+  if (playerLoaded) return true;
+  return new Promise<boolean>((resolve) => {
+    setInterval(() => {
+      if (playerLoaded) resolve(true);
+    }, 50);
+  });
+};
```

Comment:
I meant lets store the flag globally. For example, `(global as any).playerLoaded`. Then in case we need to check the load state anywhere else we can just reference `(global as any).playerLoaded` as a flag.

Reply:
The only issue is we have situations where awaiting `playerLoaded` may be necessary (proxy events being one, of course). To be honest it's a work-around, I believe it might be better to trigger a request to fetch from the client after the spawn event - but I assume that will require a bit of a rework on the NUI setup.